gobblin-config-management/gobblin-config-core/src/main/java/gobblin/config/store/hdfs/SimpleHDFSConfigStore.java

```diff
@@ -0,0 +1,341 @@
+      }
+    } catch (IOException e) {
+      throw new RuntimeException(String.format("Error while getting children for configKey: \"%s\"", configKey), e);
+    }
+    return children;
+  }
+
+  /**
+   * Retrieves all the {@link ConfigKeyPath}s that are imported by the given {@link ConfigKeyPath}. This method does this
+   * by reading the {@code includes.conf} file associated with the dataset specified by the given {@link ConfigKeyPath}.
+   *
+   * @param  configKey      the config key path whose tags are needed
+   * @param  version        the configuration version in the configuration store.
+   *
+   * @return a {@link List} of {@link ConfigKeyPath}s where each entry is a {@link ConfigKeyPath} imported by the dataset
+   * specified by the configKey.
+   *
+   * @throws VersionDoesNotExistException if the version specified cannot be found in the {@link ConfigStore}.
+   */
+  @Override
+  public List<ConfigKeyPath> getOwnImports(ConfigKeyPath configKey, String version)
+      throws VersionDoesNotExistException {
+    Preconditions.checkNotNull(configKey, "configKey cannot be null!");
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(version), "version cannot be null or empty!");
+
+    List<ConfigKeyPath> configKeyPaths = new ArrayList<>();
+    Path configDir = getDatasetDirForKey(configKey, version);
+    checkDatasetDirExists(configKey, configDir);
+
+    try {
+      for (FileStatus fileStatus : this.fs.listStatus(configDir, new IncludesConfFilterFilter())) {
+        if (!fileStatus.isDir()) {
+          try (InputStream includesConfInStream = this.fs.open(fileStatus.getPath())) {
+            configKeyPaths.addAll(Lists.newArrayList(Iterables
+                .transform(IOUtils.readLines(includesConfInStream, Charsets.UTF_8), new IncludesToConfigKey())));
+          }
+        }
+      }
+    } catch (IOException e) {
+      throw new RuntimeException(String.format("Error while getting imports for configKey: \"%s\"", configKey), e);
+    }
+    return configKeyPaths;
+  }
+
+  /**
+   * Retrieves the {@link Config} for the given {@link ConfigKeyPath} by reading the {@code main.conf} associated with
+   * the dataset specified by the given {@link ConfigKeyPath}.
+   *
+   * @param  configKey      the config key path whose properties are needed.
+   * @param  version        the configuration version in the configuration store.
+   *
+   * @return a {@link Config} for the given configKey.
+   *
+   * @throws VersionDoesNotExistException if the version specified cannot be found in the {@link ConfigStore}.
+   */
+  @Override
+  public Config getOwnConfig(ConfigKeyPath configKey, String version)
+      throws VersionDoesNotExistException {
+    Preconditions.checkNotNull(configKey, "configKey cannot be null!");
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(version), "version cannot be null or empty!");
+
+    Path configDir = getDatasetDirForKey(configKey, version);
+    checkDatasetDirExists(configKey, configDir);
```

Comment:
I'm not sure this check should be done. Querying for `/foo/bar` is valid even if `/foo/bar` doesn't exist, in that case you should just return empty?